src/rpc/claimtrie.cpp

```diff
@@ -323,6 +349,42 @@ UniValue getclaimsforname(const UniValue& params, bool fHelp)
 
     LOCK(cs_main);
     std::string name = params[0].get_str();
+
+    UniValue ret(UniValue::VOBJ);
+
+    std::vector<CBlockIndex*> indexes;
+    CCoinsViewCache coins(pcoinsTip);
+    CClaimTrieCache claims(pclaimTrie);
+    CValidationState state;
+
+    BOOST_SCOPE_EXIT(&indexes, &coins, &claims, &state) {
+        // restore the rolled-back state
+        for (std::size_t i = indexes.size(); i > 0; --i) {
+            CBlock block;
+            ReadBlockFromDisk(block, indexes.back(), Params().GetConsensus());
+            ConnectBlock(block, state, indexes[i - 1], coins, claims);
+        }
+        if (!indexes.empty()) {
+            // necessary ?
+            coins.Flush();
+            claims.flush();
+        }
+    } BOOST_SCOPE_EXIT_END
+
+    if (params.size() > 1)
+    {
+        int minconf = params[1].get_int();
+        while(--minconf > 0) {
+            indexes.push_back(chainActive.Tip());
+            CBlock block;
+            ReadBlockFromDisk(block, indexes.back(), Params().GetConsensus());
+            DisconnectBlock(block, state, indexes.back(), coins, claims);
+            // assuming we don't need to resurrect the memory pool entries for this RPC call
+        }
+        coins.Flush(); // need flushes on every iteration?
+        claims.flush();
+    }
+
     claimsForNameType claimsForName = pclaimTrie->getClaimsForName(name);
```

Comment:
There was no equivalent of `getClaimsForName` in the cache object, and it was not obvious how to achieve this method's "support" outputs using the data in the cache.